dhis-2/dhis-support/dhis-support-db-migration/src/main/resources/org/hisp/dhis/db/migration/2.39/V2_39_8__Add_soft_delete_feature_relationship.sql

```diff
@@ -0,0 +1,10 @@
+ALTER TABLE relationship ADD COLUMN IF NOT EXISTS deleted boolean DEFAULT false;
+
+ALTER TABLE relationship drop constraint fk_relationship_from_relationshipitemid;
```

Comment:
drop constraint should support `if exists` https://www.postgresql.org/docs/current/sql-altertable.html just in case the constraint is not there.

foreign key constraints also supports ALTER CONSTRAINT
but the drop with if exists/add might be more resilient if it did not exist.